circuit-chat.js

```diff
@@ -0,0 +1,280 @@
+      this.setAttribute('connected', '');
+    } else {
+      this.removeAttribute('connected');
+    }
+  }
+
+  // Get the conversation here
+  set convId(value) {
+    // If no value given or if the conversation is the same as before
+    if (!value || !value.length || value === this._convId) {
+        return;
+    }
+    this._resetValues();
+    this._convId = value;
+    // Reflect to attribute
+    this._conversationFeed.display = 'none';
+    this._input.display = 'none';
+    this.setAttribute('convId', value);
+    this._getConversation();
+  }
+
+  set maxItems(value) {
+    // Max size of conversations store at once, defaults to 100
+    this._maxItems = value ? value : 100;
+  }
+
+  set initNumOfItems(value) {
+    // Initial number of items to retrieve, defaults to 20
+    this._initNumOfItems = value ? value : 20;
+  }
+
+  get convId() {
+    return this._convId;
+  }
+
+  get client() {
+    return this._client;
+  }
+
+  constructor() {
+    super();
+    this._client = null;
+
+    this.root = this.attachShadow({ mode: 'open' });
+    this.root.appendChild(template.content.cloneNode(true));
+
+    this._usersHashMap = {}; // Hashmap to store users names for the conversation feed
+    this._conversationFeed = this.root.getElementById('conversation-feed'); // Conversation Feed
+    this._conversationTitle = this.root.getElementById('conversation-title'); //  Title of the conversation
+    this._input = this.root.querySelector('input');
+    this._btn = this.root.querySelector('button');
+  }
+
+  // Delay Circuit initialization to speed up page load. This way the SDK
+  // can be loaded with 'async'
+  async _init() {
+    // this._conversationFeed.style.overflowY = 'auto';
+    this._client = Circuit.Client({
+      domain: this._domain,
+      client_id: this._clientId
+    });
+
+    // Replacing hr from circuit items to br
+    Circuit.Injectors.itemInjector = function (item) {
+      if (item.type === Circuit.Enums.ConversationItemType.TEXT) {
+        // Replacing br and hr tags with a space
+        item.text.content = item.text.content.replace(/<(\/li|hr[\/]?)>/gi, '<br>');
+      }
+    };
+    // Event listeners
+
+    // Send a message
+    this._btn.addEventListener('click', () => this._sendTextItem());
+    this._input.addEventListener('keyup', evt => {
+      if (evt.keyCode === 13) {
+        this._sendTextItem();
+      }
+    });
+
+    // New text item added to the conversation
+    this._client.addEventListener('itemAdded', async evt => {
+      const item = evt.item;
+      if (item.convId !== this._conversation.convId || item.type !== Circuit.Enums.ConversationItemType.TEXT) {
+        return;
+      }
+      // If user isn't stored in cache get user data
+      if (!this._usersHashMap[item.creatorId]) {
+        this._usersHashMap[item.creatorId] = await this._client.getUserById(item.creatorId);
+      }
+      this._feed.push(item);
+      if (this._feed.length > this._maxItems) {
```

Comment:
I don't think we need a maxItems. If the user stays on the page, just keep adding new items.

Reply:
done